ArduCopter/mode.cpp

```diff
@@ -250,6 +250,17 @@ bool Copter::set_mode(Mode::Number mode, ModeReason reason)
         return false;
     }
 
+    // check for valid altitude if old mode did not require it but new one does
+    // we only want to stop changing modes if it could make things worse
+    if (!ignore_checks &&
+        !copter.ekf_alt_ok() &&
+        flightmode->has_manual_throttle() &&
+        !new_flightmode->has_manual_throttle()) {
+        gcs().send_text(MAV_SEVERITY_WARNING, "Mode change failed: %s requires alt", new_flightmode->name());
```

Comment:
require alt estimate ? require EKF alt ? 
"Require alt" is quite disturbing

Reply:
great, thanks, I've updated this to "needs alt estimate" which is the same wording as the pre-arm failure.